core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -49,7 +49,7 @@ object Defaults {
   val BrokerIdGenerationEnable = true
   val MaxReservedBrokerId = 1000
   val BrokerId = -1
-  val MessageMaxBytes = 1000000 + MessageSet.LogOverhead
+  val MessageMaxBytes = 1024 * 1024 + MessageSet.LogOverhead
```

Comment:
Actually just one comment. Is the producer also consistent in accounting for the message set overhead?

Reply:
I think so, but I didn't merge after Guozhang's review because I intend to write a test to ensure this is correct.

Reply:
Would be surprising if the numbers lined up perfectly since you'd also have to account for the request overhead apparently.

Reply:
We don't use request size despite the name

```
int serializedSize = AbstractRecords.estimateSizeInBytesUpperBound(apiVersions.maxUsableProduceMagic(),
                    compressionType, serializedKey, serializedValue, headers);
            ensureValidRecordSize(serializedSize);
```

Reply:
Hmm.. should we fix the config doc in that case?

Reply:
Sounds good.